henson-processor/src/main/java/com/f2prateek/dart/henson/processor/HensonExtraProcessor.java

```diff
@@ -244,12 +246,32 @@ private void parseHenson(TypeElement element, Map<TypeElement, InjectionTarget>
       TypeElement modelElement = (TypeElement) typeUtils.asElement(modelTypeMirror);
       if (!"Void".equals(modelElement.getSimpleName())) {
         if (isDebugEnabled) {
-          System.out.println(String.format("HensonNavigable class %s uses model class %s\n",
-              element.getSimpleName(), modelElement.getSimpleName()));
+          System.out.printf("HensonNavigable class %s uses model class %s\n",
+              element.getSimpleName(), modelElement.getSimpleName());
+        }
+        final InjectionTarget modelInjectionTarget;
+        //the model was processed already (it belongs to the same module and is annotated)
+        if (targetClassMap.containsKey(modelElement)) {
+          //we simply copy all extra injections from the model and add them to the target
+          modelInjectionTarget = getOrCreateTargetClass(targetClassMap, modelElement);
+        } else {
+          //otherwise, we process (again probably, this is unfortunate) the element
+          //and scan its InjectExtra annotations
+          //TODO can we avoid doing this processing again ?
+          modelInjectionTarget = scanTypeForInjectExtras(modelElement);
```

Comment:
are we using ancestors besides the parent?

Reply:
It is done on line : 270